src/hooks/loadables/useLoadRecovery.ts

```diff
@@ -0,0 +1,114 @@
+import { SENTINEL_ADDRESS } from '@safe-global/safe-core-sdk/dist/src/utils/constants'
+import { BigNumber } from 'ethers'
+import type { Delay } from '@gnosis.pm/zodiac'
+import type { TransactionAddedEvent } from '@gnosis.pm/zodiac/dist/cjs/types/Delay'
+
+import { getDelayModifiers } from '@/services/recovery/delay-modifier'
+import useAsync from '../useAsync'
+import useSafeInfo from '../useSafeInfo'
+import { useWeb3ReadOnly } from '../wallets/web3'
+import { getSpendingLimitModuleAddress } from '@/services/contracts/spendingLimitContracts'
+import useIntervalCounter from '../useIntervalCounter'
+import { useHasFeature } from '../useChains'
+import { FEATURES } from '@/utils/chains'
+import type { AsyncResult } from '../useAsync'
+import type { RecoveryQueueItem, RecoveryState } from '@/store/recoverySlice'
+
+const MAX_PAGE_SIZE = 100
+const REFRESH_DELAY = 5 * 60 * 1_000 // 5 minutes
+
+export const _getQueuedTransactionsAdded = (
+  transactionsAdded: Array<TransactionAddedEvent>,
+  txNonce: BigNumber,
+): Array<TransactionAddedEvent> => {
+  // Only queued transactions with queueNonce >= current txNonce
+  return transactionsAdded.filter(({ args }) => args.queueNonce.gte(txNonce))
+}
+
+export const _getRecoveryQueueItem = async (
+  transactionAdded: TransactionAddedEvent,
+  txCooldown: BigNumber,
+  txExpiration: BigNumber,
+): Promise<RecoveryQueueItem> => {
+  const txBlock = await transactionAdded.getBlock()
+
+  const validFrom = BigNumber.from(txBlock.timestamp).add(txCooldown)
+  const expiresAt = txExpiration.isZero()
+    ? null // Never expires
+    : validFrom.add(txExpiration)
+
+  return {
+    ...transactionAdded,
+    timestamp: txBlock.timestamp,
+    validFrom,
+    expiresAt,
+  }
+}
+
+export const _getRecoveryState = async (delayModifier: Delay): Promise<RecoveryState[number]> => {
+  const transactionAddedFilter = delayModifier.filters.TransactionAdded()
+
+  const [[modules], txExpiration, txCooldown, txNonce, queueNonce, transactionsAdded] = await Promise.all([
+    delayModifier.getModulesPaginated(SENTINEL_ADDRESS, MAX_PAGE_SIZE),
+    delayModifier.txExpiration(),
+    delayModifier.txCooldown(),
+    delayModifier.txNonce(),
+    delayModifier.queueNonce(),
+    delayModifier.queryFilter(transactionAddedFilter),
```

Comment:
I think on some networks the size of logs that can be queried is limited. 
And in general it is adviced to query the on-chain data in chunks (see [here](https://docs.chainstack.com/docs/understanding-eth-getlogs-limitations))

A good first step could be to at least limit the starting block to the block in which the Safe was deployed or the module was enabled. And for later polls we could remember the last checked block and start polling from there.

Reply:
I've created a separate [issue](https://github.com/safe-global/safe-wallet-web/issues/2792) to focus on this specifically. However, I adjusted the query logic to only request from the Safe creation block in f70fff886761a640fe434d8af3f023880fe16690. What do you think?